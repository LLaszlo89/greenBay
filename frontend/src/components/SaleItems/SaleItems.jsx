import React, { useState } from "react";
import TextField from "@material-ui/core/TextField";
import { Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Button from "@material-ui/core/Button";
import { useHistory } from "react-router-dom";
import { NavLink } from "react-router-dom";
import { addNewItem } from "../../redux/actions/itemsActions";
import { connect } from "react-redux";

const useStyles = makeStyles((theme) => ({
  root: {
    "& .MuiTextField-root": {
      margin: theme.spacing(1),
      width: "25ch",
    },
  },
  button: {
    color: "#52b202",
    fontFamily: "georgia",
  },
  link: {
    textDecoration: "none",
    fontFamily: "georgia",
    fontSize: "20px",
    color: "#006400",
    margin: "15px",
    marginTop: "10px",
    paddingTop: "5px",
  },
}));

const SaleItems = (props) => {
  const classes = useStyles();
  const history = useHistory();

  const initialState = {
    title: "",
    URL: "",
    price: null,
    description: "",
  };

  const [values, setValues] = useState(initialState);
  const [err, setErr] = useState("");

  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setValues({
      ...values,
      [name]: value,
    });
  };

  const handlerSubmitForm = async (e) => {
    e.preventDefault();
    const { title, URL, price, description } = values;
    const regex = new RegExp(    );

    if (!title) {
      setErr("All fields are required! Please fill the Title field.");
    } else if (!URL) {
      setErr("All fields are required! Please fill the URL field.");
    } else if (!URL.match(regex)) {
      setErr("Invalid URL");
    } else if (!price) {
      setErr("All fields are required! Please fill the Price field.");
    } else if (Number(price) <= 0) {
      setErr(" Price must be greater than 0 ! ");
    } else if (!description) {
      setErr("All fields are required! Please fill the description field.");
    } else {
      await props.addNew(values);
      history.push("/shop");
    }
  };

  return (
    <form className={classes.root} noValidate autoComplete="off">
      <NavLink className={classes.link} to="/shop">
        Go back to the shop
      </NavLink>
      <div>
        <Grid
          container
          spacing={2}
          direction="column"
          justify="center"
          alignItems="center"
        >
          <Typography color="error" variant="h5" align="center">
            {err}
          </Typography>
          <Grid item>
            <TextField
              id="outlined-title-input"
              label="Title"
              type="title"
              autoComplete="current-title"
              variant="outlined"
              name="title"
              value={values.title}
              onChange={handleInputChange}
              required
            />
          </Grid>
          <Grid item>
            <TextField
              id="outlined-url-input"
              label="URL"
              type="url"
              autoComplete="current-url"
              variant="outlined"
              name="URL"
              value={values.URL}
              onChange={handleInputChange}
              required
            />
          </Grid>
          <Grid item>
            <TextField
              id="outlined-price"
              label="Price"
              type="price"
              variant="outlined"
              name="price"
              value={values.price}
              onChange={handleInputChange}
              required
            />
          </Grid>
          <TextField
            id="outlined-description-input"
            label="Description"
            type="description"
            autoComplete="current-description"
            variant="outlined"
            name="description"
            value={values.description}
            onChange={handleInputChange}
            required
          />
          <Button
            variant="outlined"
            onClick={handlerSubmitForm}
            className={classes.button}
          >
            Save item
          </Button>
        </Grid>
      </div>
    </form>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    addNew: (values) => dispatch(addNewItem(values)),
  };
};

export default connect(null, mapDispatchToProps)(SaleItems);
