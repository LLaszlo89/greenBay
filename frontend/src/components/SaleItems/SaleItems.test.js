import React from "react";
import renderer from "react-test-renderer";
import SaleItems from "./SaleItems";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";

describe("Connect component to redux store ", () => {
  const mockStore = configureStore([]);

  const store = mockStore([]);

  it("SaleItems Component should render correctly", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <SaleItems />
        </Provider>
      </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
