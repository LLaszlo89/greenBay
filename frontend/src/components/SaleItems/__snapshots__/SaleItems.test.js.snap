// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect component to redux store  SaleItems Component should render correctly 1`] = `
<form
  autoComplete="off"
  className="makeStyles-root-1"
  noValidate={true}
>
  <a
    aria-current={null}
    className="makeStyles-link-3"
    href="/shop"
    onClick={[Function]}
  >
    Go back to the shop
  </a>
  <div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 MuiTypography-colorError MuiTypography-alignCenter"
      >
        
      </h5>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="outlined-title-input"
            id="outlined-title-input-label"
          >
            Title
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="current-title"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="outlined-title-input"
              name="title"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="title"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-6"
              >
                <span>
                  Title
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="outlined-url-input"
            id="outlined-url-input-label"
          >
            URL
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="current-url"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="outlined-url-input"
              name="URL"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="url"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-6"
              >
                <span>
                  URL
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormControl-root MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="outlined-price"
            id="outlined-price-label"
          >
            Price
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="outlined-price"
              name="price"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="price"
              value={null}
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-6"
              >
                <span>
                  Price
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="outlined-description-input"
          id="outlined-description-input-label"
        >
          Description
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="current-description"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="outlined-description-input"
            name="description"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="description"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-6"
            >
              <span>
                Description
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-2"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Save item
        </span>
      </button>
    </div>
  </div>
</form>
`;
